### DISCUSSION TITLE
search_all_tweets "-is:retweet" filter not working - premium academic access v2

### DISCUSSION THREAD

**Discussion by AyahAK** [tweepy/tweepy#1753] [Questions] [answered]
Hi!

I can't seem to filter out retweets using "-is:retweet". Retweets always keep showing up when I run. Can someone help?
I have premium academic access and am using api v2.

Here is my code:

query = """"lyft" OR "uber" -"RT" -is:retweet (place_country:CA OR place_country:US) lang:en"""

last_id = None
results = []

gentweets = tweepy.Paginator(client.search_all_tweets, query, since_id = last_id,
                       tweet_fields = ['id', 'text', 'created_at', 'referenced_tweets'],
                       max_results = 500)

for page in gentweets:
  tweet_data =  page.data
  if not tweet_data is None and len(tweet_data) > 0:

    for tweet in tweet_data:
      obj = {}
      obj['id'] = tweet.id
      obj['text'] = tweet.text
      obj['created_at'] = tweet.created_at
      obj['referenced_tweets'] = tweet.referenced_tweets
      results.append(obj)

  df = pd.DataFrame(results)
  df.to_csv(f"/content/drive/MyDrive/Colab Notebooks/Sentiment Analysis/Results_{len(df)}.csv")
  last_id = df.tail(1).id
  time.sleep(2)

Thank you in advance!

Ayah

**Comment by Harmon758** [comment, marked as answer]
The [order of operations](https://developer.twitter.com/en/docs/twitter-api/tweets/counts/integrate/build-a-query#order-of-operations) when evaluating your query means that the `-is:retweet` doesn't apply to the `"lyft"` query.
Something like `"(lyft OR uber) -RT -is:retweet (place_country:CA OR place_country:US) lang:en"` should work.

For code block usage, see https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks.
